docs/build/isc/v1.3/docs/tutorials/defi-yield-farming.md

```diff
@@ -0,0 +1,422 @@
+                    10 ** 18;
+
+                /// Determine reward amount, e.g., 5% of the staked value in USD
+                uint256 reward = (stakedBalanceinUSD * 5) / 100;
+
+                /// Transfer the reward to the staker
+                dappToken.transfer(
+                    recipient,
+                    reward * 10 ** dappToken.decimals()
+                );
+            }
+        }
+
+        /// Update the timestamp of the last reward issuance
+        lastReward = block.timestamp;
+    }
+}
+```
+
+## Step 3: Deployment Script (HardHat)
+
+Create a deployment script to deploy the contracts to the IOTA EVM testnet.
+
+In the `scripts` folder, create a file named `deploy.js`:
+
+```javascript
+async function main() {
+    /// Fetch contract factories
+    const DappToken = await ethers.getContractFactory("DappToken");
+    const IOTAToken = await ethers.getContractFactory("IOTAToken");
+    const TokenFarm = await ethers.getContractFactory("TokenFarm");
+
+    /// Deploy contracts
+    const dappToken = await DappToken.deploy();
+    await dappToken.deployed();
+    console.log("DappToken deployed to:", dappToken.address);
+
+    const iotaToken = await IOTAToken.deploy();
+    await iotaToken.deployed();
+    console.log("IOTAToken deployed to:", iotaToken.address);
+
+    const tokenFarm = await TokenFarm.deploy(dappToken.address, iotaToken.address, "PYTH_ORACLE_CONTRACT_ADDRESS");
+    await tokenFarm.deployed();
+    console.log("TokenFarm deployed to:", tokenFarm.address);
+}
+
+main()
+    .then(() => process.exit(0))
+    .catch((error) => {
+        console.error(error);
+        process.exit(1);
+    });
+
+```
+
+## Step 4: Deploying to the IOTA EVM Test
+
+### 1. Configure Hardhat for IOTA EVM Testnet
+
+Create a `.env` file with the following content (replace with your own private key and RPC URL):
+
+```bash
+PRIVATE_KEY=your-private-key
+IOTA_RPC_URL=https://json-rpc.evm.testnet.iotaledger.net
+```
+
+### 2. Install the dotenv package to manage environment variables:
+
+```bash
+npm install dotenv
+```
+
+### 3. Import the `dotenv` package in your `hardhat.config.js`:
+
+Modify `hardhat.config.js` to include the IOTA EVM testnet configuration:
+
+```javascript
+require("@nomicfoundation/hardhat-toolbox");
+require("dotenv").config();
+
+module.exports = {
+    solidity: "0.8.7",
+    networks: {
+        'iotaevm-testnet': {
+            url: process.env.IOTA_RPC_URL,,
+            chainId: 1075,
+            accounts: [process.env.PRIVATE_KEY], // Ensure your .env file has PRIVATE_KEY set
+        },
+    },
+};
+```
+
+## Step 5: Deploy to IOTA EVM Testnet
+
+<DeployAdmonition />
+
+### 1. Compile your contracts:
+
+```bash
+npx hardhat compile
+```
+
+### 2. Deploy your contracts to the IOTA EVM testnet:
+
+```bash
+npx hardhat run scripts/deploy.js --network evm-testnet
+```
+
+You can verify your contract by visiting
+the [EVM Testnet Explorer](https://explorer.evm.testnet.shimmer.network/),
```

Comment:
You could also write the verification command like the deploy command created above, this way few steps of verification through Block Explorer could be omitted, wdyt?

Reply:
Done, thanks.